changelog/836.feature.rst

```diff
@@ -0,0 +1,2 @@
+Introduced `to_json` method for AbstractParticleClass that converts a particle into a JSON representation
+Introduced `plasmapy.particles.from_json` method to convert JSON strings to corresponding particle
```

Comment:
Will need to be updated when issue around dumps vs dump and loads vs load is resolved.